provider/core/src/data_provider.rs

```diff
@@ -37,6 +37,20 @@ impl fmt::Display for DataRequest {
     }
 }
 
+// XXX: DataRequest should technically not implement `Borrow<ResourceOptions>` since it has
```

Comment:
@Manishearth I did this because I need a `&DataRequest` to pass down to the inner provider in the fallback provider. I can't pass ownership of the inner `ResourceOptions` to the iterator, because I need to build a `DataRequest` out of it at each step, and I can't have a local `DataRequest` and hand a `&mut ResourceOptions` to the iterator because I can't immutably borrow the `DataRequest` while one of its fields is on mutable loan. `BorrowMut` has the signature that I need, but according to docs.rs, it is not supposed to be used for borrowing a specific field, due to restrictions on `Ord` behavior.

Reply:
We're misusing BorrowMut there _already_ i think, we should be using [`AsMut`](https://doc.rust-lang.org/stable/std/convert/trait.AsMut.html)

(that's what AsRefMut is called, it's confusing, legacy stuff, sorry)